proxy/http/ForwardedConfig.cc

```diff
@@ -0,0 +1,126 @@
+/** @file
+
+  Configuration of Forwarded HTTP header option.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include <bitset>
+#include <string>
+#include <cctype>
+
+#include <ts/string_view.h>
+#include <ts/MemView.h>
+
+#include <HttpConfig.h>
+
+namespace HttpForwarded
+{
+OptionBitSet
+optStrToBitset(ts::string_view optConfigStr, std::string &error)
+{
+  using SV = ts::StringView;
+
+  const SV Delimiters(":|");
+
+  using OBS = OptionBitSet;
+
+  auto isSp = [](char c) -> bool { return std::isspace(c); };
+
+  OBS optBS;
+
+  // Convert to TS StringView to be able to use parsing members.
+  //
+  SV oCS(optConfigStr.data(), optConfigStr.size());
+
+  if (strcasecmp(oCS.trim(isSp), SV("none")) == 0) {
+    error = "";
+
+    return OBS();
+  }
+
+  do {
+    SV optStr = oCS.extractPrefix(Delimiters);
+    optStr.ltrim(isSp);
```

Comment:
For instance, here - is this really better than `optStr.ltrim(&isspace);`?

Reply:
Yes, it's better in that it compiles:

https://godbolt.org/g/zCzpqs

And, since we don't want to be memory and cpu cycle spend-thrifts, you should reconsider your infatuation with std::function.

Reply:
Using std::function results in 4 time as many lines of assembler in my example.  I don't understand what you mean about a syscall.

Some how <cctype> defines _both_  ::isspace and std::isspace.  The former converts implicitly to std::function, the latter does not.

https://godbolt.org/g/eyJf7p

Reply:
Someone answered this on stackoverflow:
https://stackoverflow.com/a/46208892/2793551

And it works in my example:
https://godbolt.org/g/mDgJpe